src/classes/GS_ApplicationStatusController.cls

```diff
@@ -0,0 +1,176 @@
+        } catch (System.NoDataFoundException ex) {
+            return Date.Today();
+        }
+    }
+
+    /**
+     * @description Retrieves the Trial expiration date.
+     * On prod this is populated quering the Organization object.
+     * For QA purpose this values can be overriden if this is running on a Scratch org and there is a value in the custom setting Testing_Configuration__c
+     * @param  overrideValuesForTesting Boolean true if usage of Testing_Configuration__c custom setting is desired
+     * @param  testingConfiguration     Map<String, Object> parsed TestingConfiguration object
+     * @param  org                      Organization the Organization object, the query should have included the TrialExpiration field 
+     * @return                          DateTime with the expirationDate.
+     */
+    private DateTime getExpirationDate(Boolean overrideValuesForTesting,Map<String,Object> testingConfiguration, Organization org) {
+        
+        if (overrideValuesForTesting) {
+            return  (testingConfiguration.get(TRIAL_EXPIRATION_PARAMETER) == null ? null : 
+                    DateTime.parse(testingConfiguration.get(TRIAL_EXPIRATION_PARAMETER).toString()));
+        } else {
+            return org.TrialExpirationDate;
+        }
+        
+    }
+
+    /**
+     * @description Returns the parsed Testing_Configuration.
+     * @return   Map<String, Object> Parsed Testing_Configuration Custom_Setting. it is a json format.
+     * Examples: 
+     * {"Sandbox":false} simulates an Active org.
+     * {"Sandbox":false,"TrialExpiration":"10/15/2020, 00:01 am”} simulates a trial org which is not applied for licenses.
+     * {"Sandbox":false,"TrialExpiration":"10/15/2020","Application":"10/02/2020"} Simulates a trial org which has applied for licenses.
+     */
+    private Map<String, Object> getTestingConfiguration() {
+        Testing_Configuration__c[] tc = [SELECT Configuration__c 
+                                         FROM   Testing_Configuration__c];
+        if (!tc.IsEmpty()) {
+            Map<String, Object> parsedConfiguration = (Map<String, Object>) JSON.deserializeUntyped(tc[0].Configuration__c);
+            return parsedConfiguration;
+        }
+        return null;
+    }
+
+    /**
+     * Returns if it's running on a Scratch org.
+     * @return true if code runs on a Scratch org, false otherwise
+     */
+    private Boolean isScratchOrg() {
+        Organization org = getOrgInformation();
+        return org.IsSandbox && org.TrialExpirationDate != null;
+    }
+
+    /** 
+     * @description Queries the Organization object to get the TrialExpirationDate and the IsSandbox field
+     * @returns Organization with TrialExpirationDate and IsSandbox queried.
+     */
+    private Organization getOrgInformation() {
+        return [SELECT TrialExpirationDate,
+                       IsSandbox
+                FROM   Organization
+                WITH_SECURITY_ENFORCED];
```

Comment:
I'm not so sure about using `WITH_SECURITY_ENFORCED` here. This is a system object which technically nobody probably access to. We recently had an issue where standard users were getting exceptions thrown because code running as them was querying `AsyncApexJob`, which I consider at the same system level as this object. Have you verified that a standard user (without any system admin level permissions) can still use this page without error?

Reply:
This component is going to be seen only by System Administrators, so the WITH_SECURITY_ENFORCED should not be an Issue. I also wrote a test with Run As standard user and the fields were visible. I'm pretty sure that WITH_SECURITY_ENFORCED would cause no harm on this one and I avoid a PMD warning, but it can be removed if required.

Reply:
Since this is a system object, where it's not even possible to set security on this object, I recommend removing the `WITH SECURITY ENFORCED` clause in the query.